cmd/troubleshoot/cli/util.go

```diff
@@ -27,16 +26,15 @@ func isURL(str string) bool {
 	return true
 }
 
-func createTroubleshootK8sClient() (*troubleshootclientv1beta1.TroubleshootV1beta1Client, error) {
-	v := viper.GetViper()
-
-	config, err := clientcmd.BuildConfigFromFlags("", v.GetString("kubecontext"))
+func createTroubleshootK8sClient(configFlags *genericclioptions.ConfigFlags) (*troubleshootclientv1beta1.TroubleshootV1beta1Client, error) {
+	config, err := configFlags.ToRESTConfig()
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "read kubeconfig")
 	}
+
 	troubleshootClient, err := troubleshootclientv1beta1.NewForConfig(config)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "create client")
```

Comment:
"create client" . -- will you make this a message?  "failed to create troubleshootclient"